File: ios/Sasquatsh/Views/Billing/BillingView.swift
import SwiftUI

struct BillingView: View {
    @Environment(\.services) private var services
    @Environment(\.openURL) private var openURL
    @Environment(AuthViewModel.self) private var authVM
    @State private var vm = BillingViewModel()
    @State private var showCancelConfirm = false
    @State private var selectedInvoice: Invoice?

    private var storeKit: StoreKitService {
        services.storeKit
    }

    var body: some View {
        ScrollView {
            if vm.isLoading && vm.subscriptionInfo == nil {
                LoadingView()
            } else {
                VStack(spacing: 20) {
                    if let error = vm.error {
                        ErrorBannerView(message: error) { vm.error = nil }
                    }

                    if let msg = vm.successMessage {
                        Text(msg)
                            .font(.md3BodyMedium)
                            .foregroundStyle(.green)
                            .padding(.horizontal)
                    }

                    currentPlanCard

                    NavigationLink {
                        PricingView()
                    } label: {
                        HStack {
                            Image(systemName: "sparkles")
                                .foregroundStyle(Color.md3Primary)
                            Text("View All Plans")
                                .font(.md3LabelLarge)
                                .foregroundStyle(Color.md3Primary)
                            Spacer()
                            Image(systemName: "chevron.right")
                                .font(.md3BodySmall)
                                .foregroundStyle(Color.md3OnSurfaceVariant)
                        }
                        .padding()
                        .cardStyle()
                    }

                    // Only show Stripe payment/invoice sections for Stripe subscriptions
                    if vm.isStripeSubscription || (!vm.isAppleSubscription && !storeKit.hasActiveSubscription) {
                        paymentMethodCard
                        invoiceHistoryCard
                    }
                }
                .padding()
            }
        }
        .background(Color.md3SurfaceContainer)
        .navigationTitle("Billing")
        .navigationBarTitleDisplayMode(.inline)
        .refreshable {
            await vm.loadBillingInfo()
            await storeKit.updatePurchasedProducts()
        }
        .confirmationDialog("Cancel Subscription", isPresented: $showCancelConfirm) {
            Button("Cancel Subscription", role: .destructive) {
                Task { await vm.cancelSubscription() }
            }
        } message: {
            Text("Your subscription will remain active until the end of the current billing period.")
        }
        .sheet(item: $selectedInvoice) { invoice in
            InvoiceDetailSheet(invoice: invoice, viewModel: vm)
        }
        .task {
            vm.configure(services: services)
            await vm.loadBillingInfo()
            await storeKit.updatePurchasedProducts()
        }
    }

    // MARK: - Current Plan Card

    private var effectiveTier: SubscriptionTier {
        // Check StoreKit entitlements first, then server tier
        storeKit.currentTier ?? authVM.user?.effectiveTier ?? vm.currentTier
    }

    private var isApple: Bool {
        storeKit.hasActiveSubscription || vm.isAppleSubscription
    }

    private var currentPlanCard: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text("Current Plan")
                    .font(.md3TitleMedium)
                    .foregroundStyle(Color.md3OnSurface)
                Spacer()
                SubscriptionBadgeView(tier: effectiveTier)
            }

            // Price
            if isApple, let productId = storeKit.purchasedProductIDs.first,
               let product = storeKit.products.first(where: { $0.product.id == productId }) {
                Text("\(product.product.displayPrice)/\(product.period == .annual ? "year" : "mo")")
                    .font(.md3HeadlineSmall)
                    .foregroundStyle(Color.md3Primary)
            } else {
                Text(effectiveTier.priceLabel)
                    .font(.md3HeadlineSmall)
                    .foregroundStyle(Color.md3Primary)
            }

            // Subscription source
            if isApple {
                HStack(spacing: 6) {
                    Image(systemName: "apple.logo")
                        .font(.md3BodySmall)
                    Text("Managed by Apple")
                        .font(.md3BodySmall)
                }
                .foregroundStyle(Color.md3OnSurfaceVariant)
            } else if vm.isStripeSubscription {
                HStack(spacing: 6) {
                    Image(systemName: "creditcard.fill")
                        .font(.md3BodySmall)
                    Text("Managed by Stripe")
                        .font(.md3BodySmall)
                }
                .foregroundStyle(Color.md3OnSurfaceVariant)
            }

            if vm.hasOverride {
                HStack(spacing: 4) {
                    Image(systemName: "star.fill")
                        .font(.md3BodySmall)
                        .foregroundStyle(Color.md3Tertiary)
                    Text("Admin override active")
                        .font(.md3BodySmall)
                        .foregroundStyle(Color.md3OnSurfaceVariant)
                }
            }

            // Features checklist
            VStack(alignment: .leading, spacing: 6) {
                ForEach(TierConfig.tierFeatureDescriptions(for: effectiveTier), id: \.self) { feature in
                    HStack(spacing: 6) {
                        Image(systemName: "checkmark.circle.fill")
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3Primary)
                        Text(feature)
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                    }
                }
            }

            Divider()

            // Action buttons
            if isApple {
                // Apple subscription management
                Button {
                    openURL(vm.manageAppleSubscriptionURL)
                } label: {
                    HStack {
                        Image(systemName: "apple.logo")
                        Text("Manage in App Store")
                    }
                    .outlinedButtonStyle()
                }

                Button {
                    Task {
                        try? await storeKit.restorePurchases()
                        await authVM.refreshUser()
                    }
                } label: {
                    Text("Restore Purchases")
                        .font(.md3LabelLarge)
                        .foregroundStyle(Color.md3Primary)
                        .frame(maxWidth: .infinity)
                        .frame(height: 36)
                }
            } else if effectiveTier == .free && !vm.hasOverride {
                NavigationLink {
                    PricingView()
                } label: {
                    Text("Upgrade Plan")
                        .primaryButtonStyle()
                }
            } else if vm.isCancelled {
                Button {
                    Task { await vm.reactivateSubscription() }
                } label: {
                    Text("Reactivate Subscription")
                        .primaryButtonStyle()
                }
                .disabled(vm.actionLoading)

                if let expiresAt = vm.subscriptionInfo?.subscription.expiresAt {
                    Text("Access until \(vm.formattedDate(expiresAt))")
                        .font(.md3BodySmall)
                        .foregroundStyle(Color.md3OnSurfaceVariant)
                }
            } else if vm.hasActiveSubscription && vm.isStripeSubscription {
                HStack(spacing: 12) {
                    NavigationLink {
                        PricingView()
                    } label: {
                        Text("Change Plan")
                            .outlinedButtonStyle()
                    }

                    Button {
                        showCancelConfirm = true
                    } label: {
                        Text("Cancel")
                            .font(.md3LabelLarge)
                            .foregroundStyle(Color.md3Error)
                            .frame(maxWidth: .infinity)
                            .frame(height: 40)
                    }
                    .disabled(vm.actionLoading)
                }
            }

            if vm.isPastDue {
                HStack(spacing: 4) {
                    Image(systemName: "exclamationmark.triangle.fill")
                        .foregroundStyle(Color.md3Error)
                    Text("Payment past due — please update your payment method")
                        .font(.md3BodySmall)
                        .foregroundStyle(Color.md3Error)
                }
            }
        }
        .padding()
        .cardStyle()
    }

    // MARK: - Payment Method Card (Stripe only)

    private var paymentMethodCard: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text("Payment Method")
                .font(.md3TitleMedium)
                .foregroundStyle(Color.md3OnSurface)

            if let pm = vm.subscriptionInfo?.paymentMethod {
                HStack {
                    Image(systemName: "creditcard.fill")
                        .foregroundStyle(Color.md3Primary)
                    VStack(alignment: .leading, spacing: 2) {
                        Text("\(vm.formatCardBrand(pm.brand)) ••••\(pm.last4)")
                            .font(.md3BodyMedium)
                        Text("Expires \(pm.expMonth)/\(pm.expYear)")
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                    }
                    Spacer()
                }

                Button {
                    openURL(AppConfig.billingURL)
                } label: {
                    Text("Manage")
                        .outlinedButtonStyle()
                }
            } else {
                Text("No payment method on file")
                    .font(.md3BodyMedium)
                    .foregroundStyle(Color.md3OnSurfaceVariant)
            }
        }
        .padding()
        .cardStyle()
    }

    // MARK: - Invoice History Card (Stripe only)

    private var invoiceHistoryCard: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text("Invoice History")
                .font(.md3TitleMedium)
                .foregroundStyle(Color.md3OnSurface)

            if vm.invoices.isEmpty {
                Text("No invoices yet")
                    .font(.md3BodyMedium)
                    .foregroundStyle(Color.md3OnSurfaceVariant)
                    .padding(.vertical, 8)
            } else {
                ForEach(vm.invoices) { invoice in
                    Button {
                        selectedInvoice = invoice
                    } label: {
                        HStack {
                            VStack(alignment: .leading, spacing: 2) {
                                Text(vm.formattedDate(invoice.invoiceDate))
                                    .font(.md3BodyMedium)
                                    .foregroundStyle(Color.md3OnSurface)
                                Text(vm.formattedAmount(invoice.amountCents))
                                    .font(.md3BodySmall)
                                    .foregroundStyle(Color.md3OnSurfaceVariant)
                            }
                            Spacer()
                            invoiceStatusBadge(invoice.status)
                            Image(systemName: "chevron.right")
                                .font(.md3BodySmall)
                                .foregroundStyle(Color.md3OnSurfaceVariant)
                        }
                        .padding(.vertical, 4)
                    }
                    .buttonStyle(.plain)
                }

                if vm.hasMoreInvoices {
                    Button {
                        Task { await vm.loadMoreInvoices() }
                    } label: {
                        Text("Load More")
                            .font(.md3LabelLarge)
                            .foregroundStyle(Color.md3Primary)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 8)
                    }
                }
            }
        }
        .padding()
        .cardStyle()
    }

    private func invoiceStatusBadge(_ status: InvoiceStatus) -> some View {
        let (text, color): (String, Color) = switch status {
        case .paid: ("Paid", .md3PrimaryContainer)
        case .open: ("Open", .md3TertiaryContainer)
        case .draft: ("Draft", .md3SurfaceContainerHigh)
        case .void: ("Void", .md3ErrorContainer)
        case .uncollectible: ("Uncollectible", .md3ErrorContainer)
        }
        return BadgeView(text: text, color: color)
    }
}

File: android/app/src/main/java/com/sasquatsh/app/views/billing/BillingView.kt
package com.sasquatsh.app.views.billing

import android.content.Intent
import android.net.Uri
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import com.sasquatsh.app.config.AppConfig
import com.sasquatsh.app.models.InvoiceStatus
import com.sasquatsh.app.models.SubscriptionTier
import com.sasquatsh.app.viewmodels.AuthViewModel
import com.sasquatsh.app.viewmodels.BillingViewModel
import com.sasquatsh.app.views.profile.SubscriptionBadge
import com.sasquatsh.app.views.shared.BadgeView
import com.sasquatsh.app.views.shared.D20SpinnerView
import com.sasquatsh.app.views.shared.ErrorBannerView

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun BillingView(
    authViewModel: AuthViewModel,
    onNavigateToPricing: () -> Unit,
    onBack: () -> Unit,
    billingViewModel: BillingViewModel = hiltViewModel()
) {
    val uiState by billingViewModel.uiState.collectAsState()
    val authState by authViewModel.uiState.collectAsState()
    val context = LocalContext.current
    var showCancelConfirm by rememberSaveable { mutableStateOf(false) }

    LaunchedEffect(Unit) {
        billingViewModel.loadBillingInfo()
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Billing") },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = "Back"
                        )
                    }
                }
            )
        }
    ) { padding ->
        if (uiState.isLoading && uiState.subscriptionInfo == null) {
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(padding),
                contentAlignment = Alignment.Center
            ) {
                D20SpinnerView(size = 40.dp, modifier = Modifier.size(40.dp))
            }
        } else {
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(padding)
                    .background(MaterialTheme.colorScheme.surfaceContainer)
                    .verticalScroll(rememberScrollState())
                    .padding(16.dp),
                verticalArrangement = Arrangement.spacedBy(20.dp)
            ) {
                // Error banner
                uiState.error?.let { error ->
                    ErrorBannerView(
                        message = error,
                        onDismiss = { billingViewModel.clearMessages() }
                    )
                }

                // Success message
                uiState.successMessage?.let { msg ->
                    Text(
                        text = msg,
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.primary
                    )
                }

                // Current Plan Card
                CurrentPlanCard(
                    billingViewModel = billingViewModel,
                    authViewModel = authViewModel,
                    onNavigateToPricing = onNavigateToPricing,
                    onCancelClick = { showCancelConfirm = true }
                )

                // View All Plans
                Card(
                    onClick = onNavigateToPricing,
                    shape = RoundedCornerShape(12.dp),
                    colors = CardDefaults.cardColors(
                        containerColor = MaterialTheme.colorScheme.surfaceContainer
                    )
                ) {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(16.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(
                            imageVector = Icons.Default.AutoAwesome,
                            contentDescription = null,
                            tint = MaterialTheme.colorScheme.primary
                        )
                        Spacer(modifier = Modifier.width(8.dp))
                        Text(
                            text = "View All Plans",
                            style = MaterialTheme.typography.labelLarge,
                            color = MaterialTheme.colorScheme.primary,
                            modifier = Modifier.weight(1f)
                        )
                        Icon(
                            imageVector = Icons.Default.ChevronRight,
                            contentDescription = null,
                            tint = MaterialTheme.colorScheme.onSurfaceVariant,
                            modifier = Modifier.size(16.dp)
                        )
                    }
                }

                // Payment Method Card (Stripe only)
                if (uiState.isStripeSubscription || (!uiState.isGoogleSubscription)) {
                    PaymentMethodCard(billingViewModel = billingViewModel)
                    InvoiceHistoryCard(billingViewModel = billingViewModel)
                }
            }
        }
    }

    // Cancel confirmation
    if (showCancelConfirm) {
        AlertDialog(
            onDismissRequest = { showCancelConfirm = false },
            title = { Text("Cancel Subscription") },
            text = {
                Text("Your subscription will remain active until the end of the current billing period.")
            },
            confirmButton = {
                TextButton(
                    onClick = {
                        billingViewModel.cancelSubscription(context)
                        showCancelConfirm = false
                    },
                    colors = ButtonDefaults.textButtonColors(
                        contentColor = MaterialTheme.colorScheme.error
                    )
                ) {
                    Text("Cancel Subscription")
                }
            },
            dismissButton = {
                TextButton(onClick = { showCancelConfirm = false }) {
                    Text("Keep Subscription")
                }
            }
        )
    }
}

@Composable
private fun CurrentPlanCard(
    billingViewModel: BillingViewModel,
    authViewModel: AuthViewModel,
    onNavigateToPricing: () -> Unit,
    onCancelClick: () -> Unit
) {
    val uiState by billingViewModel.uiState.collectAsState()
    val authState by authViewModel.uiState.collectAsState()
    val effectiveTier = authState.user?.effectiveTier ?: uiState.currentTier
    val context = LocalContext.current

    Card(
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceContainer
        )
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            // Header
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Current Plan",
                    style = MaterialTheme.typography.titleMedium,
                    color = MaterialTheme.colorScheme.onSurface
                )
                SubscriptionBadge(tier = effectiveTier)
            }

            // Price
            Text(
                text = effectiveTier.priceLabel,
                style = MaterialTheme.typography.headlineSmall,
                color = MaterialTheme.colorScheme.primary
            )

            // Source
            if (uiState.isGoogleSubscription) {
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(6.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.ShoppingCart,
                        contentDescription = null,
                        modifier = Modifier.size(14.dp),
                        tint = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                    Text(
                        text = "Managed by Google Play",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            } else if (uiState.isStripeSubscription) {
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(6.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.CreditCard,
                        contentDescription = null,
                        modifier = Modifier.size(14.dp),
                        tint = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                    Text(
                        text = "Managed by Stripe",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }

            // Override indicator
            if (uiState.hasOverride) {
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(4.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.Star,
                        contentDescription = null,
                        modifier = Modifier.size(14.dp),
                        tint = MaterialTheme.colorScheme.tertiary
                    )
                    Text(
                        text = "Admin override active",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }

            // Features checklist
            val features = tierFeatures(effectiveTier)
            Column(verticalArrangement = Arrangement.spacedBy(6.dp)) {
                features.forEach { feature ->
                    Row(
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(6.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Default.CheckCircle,
                            contentDescription = null,
                            tint = MaterialTheme.colorScheme.primary,
                            modifier = Modifier.size(16.dp)
                        )
                        Text(
                            text = feature,
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                }
            }

            HorizontalDivider()

            // Action buttons
            if (uiState.isGoogleSubscription) {
                OutlinedButton(
                    onClick = {
                        val intent = Intent(
                            Intent.ACTION_VIEW,
                            uiState.manageGoogleSubscriptionUri
                        )
                        context.startActivity(intent)
                    },
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(24.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.ShoppingCart,
                        contentDescription = null,
                        modifier = Modifier.size(18.dp)
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Text("Manage in Google Play")
                }
            } else if (effectiveTier == SubscriptionTier.FREE && !uiState.hasOverride) {
                Button(
                    onClick = onNavigateToPricing,
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(24.dp)
                ) {
                    Text("Upgrade Plan")
                }
            } else if (uiState.isCancelled) {
                Button(
                    onClick = { billingViewModel.reactivateSubscription() },
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(24.dp),
                    enabled = !uiState.actionLoading
                ) {
                    Text("Reactivate Subscription")
                }

                uiState.subscriptionInfo?.subscription?.expiresAt?.let { expiresAt ->
                    Text(
                        text = "Access until ${billingViewModel.formattedDate(expiresAt)}",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            } else if (uiState.hasActiveSubscription && uiState.isStripeSubscription) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    OutlinedButton(
                        onClick = onNavigateToPricing,
                        modifier = Modifier.weight(1f),
                        shape = RoundedCornerShape(24.dp)
                    ) {
                        Text("Change Plan")
                    }
                    TextButton(
                        onClick = onCancelClick,
                        enabled = !uiState.actionLoading,
                        colors = ButtonDefaults.textButtonColors(
                            contentColor = MaterialTheme.colorScheme.error
                        )
                    ) {
                        Text("Cancel")
                    }
                }
            }

            // Past due warning
            if (uiState.isPastDue) {
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(4.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.Warning,
                        contentDescription = null,
                        tint = MaterialTheme.colorScheme.error,
                        modifier = Modifier.size(16.dp)
                    )
                    Text(
                        text = "Payment past due -- please update your payment method",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.error
                    )
                }
            }
        }
    }
}

@Composable
private fun PaymentMethodCard(billingViewModel: BillingViewModel) {
    val uiState by billingViewModel.uiState.collectAsState()
    val context = LocalContext.current

    Card(
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceContainer
        )
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Text(
                text = "Payment Method",
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onSurface
            )

            val pm = uiState.subscriptionInfo?.paymentMethod
            if (pm != null) {
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.CreditCard,
                        contentDescription = null,
                        tint = MaterialTheme.colorScheme.primary
                    )
                    Column {
                        Text(
                            text = "${billingViewModel.formatCardBrand(pm.brand)} ****${pm.last4}",
                            style = MaterialTheme.typography.bodyMedium
                        )
                        Text(
                            text = "Expires ${pm.expMonth}/${pm.expYear}",
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                }

                OutlinedButton(
                    onClick = {
                        val intent = Intent(
                            Intent.ACTION_VIEW,
                            Uri.parse(AppConfig.BILLING_URL)
                        )
                        context.startActivity(intent)
                    },
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(24.dp)
                ) {
                    Text("Manage")
                }
            } else {
                Text(
                    text = "No payment method on file",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}

@Composable
private fun InvoiceHistoryCard(billingViewModel: BillingViewModel) {
    val uiState by billingViewModel.uiState.collectAsState()

    Card(
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceContainer
        )
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Text(
                text = "Invoice History",
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onSurface
            )

            if (uiState.invoices.isEmpty()) {
                Text(
                    text = "No invoices yet",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(vertical = 8.dp)
                )
            } else {
                uiState.invoices.forEach { invoice ->
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(vertical = 4.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column(modifier = Modifier.weight(1f)) {
                            Text(
                                text = billingViewModel.formattedDate(invoice.invoiceDate),
                                style = MaterialTheme.typography.bodyMedium,
                                color = MaterialTheme.colorScheme.onSurface
                            )
                            Text(
                                text = billingViewModel.formattedAmount(invoice.amountCents),
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                        InvoiceStatusBadge(status = invoice.status)
                        Spacer(modifier = Modifier.width(8.dp))
                        Icon(
                            imageVector = Icons.Default.ChevronRight,
                            contentDescription = null,
                            tint = MaterialTheme.colorScheme.onSurfaceVariant,
                            modifier = Modifier.size(16.dp)
                        )
                    }
                }

                if (uiState.hasMoreInvoices) {
                    TextButton(
                        onClick = { billingViewModel.loadMoreInvoices() },
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Text(
                            text = "Load More",
                            style = MaterialTheme.typography.labelLarge,
                            color = MaterialTheme.colorScheme.primary
                        )
                    }
                }
            }
        }
    }
}

@Composable
private fun InvoiceStatusBadge(status: InvoiceStatus) {
    val (text, color) = when (status) {
        InvoiceStatus.PAID -> "Paid" to MaterialTheme.colorScheme.primaryContainer
        InvoiceStatus.OPEN -> "Open" to MaterialTheme.colorScheme.tertiaryContainer
        InvoiceStatus.DRAFT -> "Draft" to MaterialTheme.colorScheme.surfaceContainerHigh
        InvoiceStatus.VOID -> "Void" to MaterialTheme.colorScheme.errorContainer
        InvoiceStatus.UNCOLLECTIBLE -> "Uncollectible" to MaterialTheme.colorScheme.errorContainer
    }
    BadgeView(text = text, color = color)
}

private fun tierFeatures(tier: SubscriptionTier): List<String> {
    return when (tier) {
        SubscriptionTier.FREE -> listOf(
            "Host 1 game per event",
            "Create 1 group",
            "Basic event management",
            "Join unlimited events"
        )
        SubscriptionTier.BASIC -> listOf(
            "Up to 5 games per event",
            "Create up to 5 groups",
            "1 recurring game per group",
            "Table/room/hall locations",
            "Game night planning",
            "Event chat",
            "No ads"
        )
        SubscriptionTier.PRO -> listOf(
            "Up to 10 games per event",
            "Create up to 10 groups",
            "Unlimited recurring games",
            "Table/room/hall locations",
            "Game night planning",
            "Items to bring lists",
            "Event chat",
            "No ads"
        )
        SubscriptionTier.PREMIUM -> listOf(
            "All Pro features",
            "Priority support",
            "Custom branding"
        )
    }
}
